lib/src/main/java/com/dynatrace/metric/util/Metric.java

```diff
@@ -336,12 +339,51 @@ public String serialize() throws MetricException {
       return builder.toString();
     }
 
-    private String makeNormalizedMetricKey() {
+    public String serializeMetadataLine() throws MetricException {
+      // getNormalizedMetricKey will normalize the key if not already normalized, and throw if the
+      // key is invalid
+      String normalizedKeyString = getNormalizedMetricKey();
+
+      if (this.value == null) {
+        throw new MetricException("No value set for metric.");
+      }
+
+      return MetadataLineFactory.createMetadataLine(
+          normalizedKeyString, description, unit, this.value.getMetricType().toString());
+    }
+
+    /**
+     * @return A {@link String} containing all properties set on the {@link Metric.Builder} in an
+     *     ingestion-ready format.
+     * @throws MetricException If no value is set or if the prefix/metric key combination evaluates
+     *     to an invalid/empty metric key after normalization. Will also throw a {@link
+     *     MetricException} when the line length after serialization exceeds the maximum line length
+     *     accepted by the ingest API.
+     * @deprecated please use {@link Metric.Builder#serializeMetricLine()} Serialize a {@link
+     *     Metric.Builder} object to a String in a valid format for the Dynatrace metrics ingest
+     *     endpoint.
+     */
+    @Deprecated
+    public String serialize() throws MetricException {
+      return serializeMetricLine();
+    }
+
+    public String getNormalizedMetricKey() throws MetricException {
```

Comment:
I guess it can, I wonder if we want to keep it public though. It should not increase the overhead, as we will have to call it at some point anyway, but it might be helpful in other situations (e.g. when you need to keep track of previously seen metric keys)

Reply:
Sure. We just need to make sure we will not be impacted by the fact people will rely on it. But if we already see value in it being public, I'm fine with it.